gems/sorbet-runtime/lib/types/types/simple.rb

```diff
@@ -29,5 +29,27 @@ def valid?(obj)
         false
       end
     end
+
+    def to_nilable
+      @nilable ||= T::Types::Union.new([self, T::Utils::Nilable::NIL_TYPE])
+    end
+
+    class << self
+      alias_method :unpooled, :new
+      private :new
+    end
+
+    module Private
+      module Pool
+        def self.type_for_module(mod)
+          cached = mod.instance_variable_get(:@__as_sorbet_simple_type)
+          return cached if cached
+
+          type = Simple.unpooled(mod)
+          mod.instance_variable_set(:@__as_sorbet_simple_type, type) unless mod.frozen?
```

Comment:
Can you add a test using frozen modules?

Specifically:

- That it doesn't raise when calling this method
- That the resulting types are not reference equal